src/server/hook_func.c

```diff
@@ -993,7 +993,7 @@ mgr_hook_delete(struct batch_request *preq)
 		return;
 	}
 
-	hookname = preq->rq_ind.rq_manager.rq_objname;
+	strncpy(hookname, preq->rq_ind.rq_manager.rq_objname, PBS_MAXSVRJOBID);
```

Comment:
Perhaps doing a strdup and freeing after using is good enough?

Reply:
I can change it to strdup.
But just for my knowledge why strdup instead strncpy? What we will gain by using strdup?

Reply:
No additional gain. The are essentially the same. You declare a local variable and have to carefully determine the size and do a proper strncpy and ensure it is null terminated. Besides making sure that the size variable makes sense and does not confuse people (due to it's name  referring to jobid size). A strdup is easy, faster than strncpy in many cases and does the same work, and is less code. Just a suggestion, choose what you like best.